src/bpftrace.cpp

```diff
@@ -962,13 +950,30 @@ int BPFtrace::run(BpfBytecode bytecode)
     }
   }
 
-  // XXX: cast is a workaround for a gcc bug that causes an "invalid conversion"
-  // error here in case the trigger function has the `target` attribute set (see
-  // triggers.h): https://gcc.godbolt.org/z/44b5MjdbT
-  if (run_special_probe("BEGIN_trigger",
-                        bytecode_,
-                        reinterpret_cast<trigger_fn_t>(BEGIN_trigger)))
-    return -1;
+  for (auto &special_probe :
+       std::ranges::reverse_view(resources.special_probes)) {
+    if (special_probe.attach_point == "BEGIN_trigger") {
```

Comment:
It looks like it's time to start storing special probes in a map hashed by the probe name. That could remove these loops and simplify the code a lot.

Reply:
how about using a function instead? if we store special probes in a map, in different parts across the code (E.g [this](https://github.com/bpftrace/bpftrace/blob/57b355c95b5406742edeb4b99330e2253703615b/src/bpfbytecode.cpp#L190)) we need to use templates or convert the std::map to a std::vec.

Reply:
Sure, it was just an idea. We're talking units of probes here so storing them in a map will not bring any real performance benefits and, as you notice, complicate other code paths. Function sounds like a way to go, the only problem is that sometimes we need to *attach and trigger* a probe (BEGIN, END) and sometimes *just attach* it (signals).